packages/pg-vector-selector/src/utils.ts

```diff
@@ -0,0 +1,34 @@
+export function getRetry(count: number) {
+  if (count < 1) {
+    throw new Error("count should be greater than 0");
+  }
```

Comment:
[IMO] How about this
- if `count` is a negative integer, throw an error
- if `count` is 0, it just run `return await fn();` on top of the `getRetry` implementation
- if `count` is more than 0, let's go!!!

This follow is more testable I think(Also we have `NonNegativeInteger` type, so It is more convenient to give special treatment to the case of 0.